examples/nlp/word_language_model/model_def.py

```diff
@@ -0,0 +1,152 @@
+"""
+This example is to show how to use an the PyTorch Word Language Modeling example with Determined.
+The flags and configurations can be found under const.yaml for single GPU training, and distributed.yaml
+for distributed training across 8 GPUs. For more information
+regarding the optional flags view the original script linked below.
+This implementation is based on:
+https://github.com/pytorch/examples/tree/master/word_language_model
+"""
+from typing import Dict, Sequence, Union, Any
+import torch
+import torch.nn as nn
+import time
+from pathlib import Path
+
+from determined.pytorch import (
+    DataLoader,
+    PyTorchTrial,
+    PyTorchTrialContext,
+    LRScheduler,
+)
+import data
+from model import TransformerModel, RNNModel
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+
+
+class WordLanguageModelPyTorch(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext):
+        self.context = context
+        data_config = self.context.get_data_config()
+        hparams = self.context.get_hparams()
+        self.using_bind_mount = data_config.get("use_bind_mount", False)
```

Comment:
For this, my reasoning was that I didn't want users to have to specify all of the numbers, especially since many of them have defaults (the original values in the pytorch repo). I can go ahead and specify them in the .yaml files if you'd prefer, instead!

Reply:
You can just set the numbers in the .yaml files though right? Either way, the users would have to specify them. I think it makes for cleaner code to have all the config settings reside in the .yaml so let's do that.